schema.c

```diff
@@ -4096,6 +4107,55 @@ sch_traverse_tree (sch_instance * instance, sch_node * schema, GNode * node, int
     return rc;
 }
 
+static void
```

Comment:
I think comments would be helpful here - I am a bit puzzled that you are doing recursive descent and deep copying?

Reply:
What is happening is that tree2 is being merged into tree1. The deep copying instead on moving nodes between the trees means it is much cleaner to delete tree2 when the merge is complete. I will add a comment.